expo/expo
# [expo-dev-menu] update dev menu settings when initial props change

# Why

resolves an issue discovered when testing #17650 -- the devSettings were not updating when the initial props changed

# How

use an effect to update the `devSettings` state when the `initialDevSettings` (from context) change

# Test Plan

- open a locally served app in development, turn on remote debugging
- toggle it off and on a few times, make sure the setting is correct in the dev menu each time
- go back to launcher
- open the same update again
- open dev menu, see that remote debugging is off
- turn it on again, should work

- try toggling fast refresh as well, make sure the correct state shows in the dev menu each time

# Checklist
